doc/src/manual/embedding.md

```diff
@@ -284,6 +285,62 @@ args[1] = some_other_value;
 JL_GC_POP();
 ```
 
+Each scope must have only one call to `JL_GC_PUSH*`. Hence, if all variables cannot be pushed once by
+a single call to `JL_GC_PUSH*`, or if there is more than 6 variables to be pushed and using an array
+of arguments is not an option, then one can use inner blocks:
+
+```c
+jl_value_t *ret1 = jl_eval_string("sqrt(2.0)");
+JL_GC_PUSH1(&ret1);
+jl_value_t *ret2 = 0;
+{
+    jl_function_t *func = jl_get_function(jl_base_module, "exp");
+    ret2 = jl_call1(func, ret1);
+    JL_GC_PUSH1(&ret2);
+    // Do something with ret2.
+    JL_GC_POP();    // This pops ret2.
+}
+JL_GC_POP();    // This pops ret1.
+```
+
+If it is required to hold the pointer to a variable between functions (or block scopes), then it is
+not possible to use `JL_GC_PUSH*`. In this case, it is necessary to create and keep a reference to the
+variable in the Julia global scope. One simple way to accomplish this is to use a global `IdDict` that
+will hold the references, avoiding deallocation by the GC. However, this method will only work
+properly with mutable types. If the variable is immutable, then it needs to be wrapped in an
+equivalent mutable container (like `Vector`) before it is pushed to `IdDict`. In this approach, the C
```

Comment:
As I said, if you want to wrap it, a `RefValue{Any}` is much better at doing that.

Reply:
Ok! I will update to advice the user to use `RefValue` (`Ref(var)` right?) instead of `Vector`.